### DISCUSSION TITLE
Can you detect from which app a link was clicked?

### DISCUSSION THREAD

**Discussion by brandonrdnt** [U-C-S/Hurl#61] [Ideas]
Hello!  I found your app while looking for a Windows alternative to Velja (https://apps.apple.com/us/app/velja/id1607635845?mt=12), an indispensable app that I use on my Macs.  The main workflow that I use Velja for is to open URLs in a specific browser based on the app in which I clicked on the link.  For example, any links I open in Microsoft Outlook or in Microsoft Teams are set to open in Firefox, but every other link goes to Edge.  

I'm a 10+ year C# developer and I'm considering attempting to implement this feature in Hurl (I'd love to PR it back into your project if it works) but I wanted to ask before I attempted, are you aware of any way in Windows to know the source application from which a URL was clicked?  Is this data provided anywhere in the URL-handling process?

Maybe you've already implemented this and I just don't know it :) But I thought I'd check either way.

Thank you!

**Comment by U-C-S** [comment]
Great idea. Though I'm not aware of any such way, and the closest i could think off is identifying the previous active process (before Hurl becomes the active window), which may or maynot be the source application for URL click! Which I'm not much sure about implementation.

Also URL handling currently only takes care of few parameters as written in this [file](https://github.com/U-C-S/Hurl/blob/main/App/Helpers/CliArgs.cs)
PR is welcome if u any working or even prototype solution.

**Comment by vnaumenko** [comment]
Maybe choose a browser by the link (pattern)?
I have a browser for work, and all links with the company's domain open in a working browser.
Youtube, Twitter and other opens in a home browser.

**Reply by U-C-S** [reply]
Yeh the one u described is #18 and currently in development
